src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,534 @@
+                var uuidMedication = Uuid.Generate().Value;
+                var medicationId = uuid.Split(":").Last();
+                var medicationRequestRepresentation = new MedicationRequestRepresentation
+                {
+                    FullUrl = uuid,
+                    Resource = new MedicationRequestResource
+                    {
+                        Id = id,
+                        Intent = "order",
+                        Status = "active",
+                        Subject = new Subject(patientName),
+                        AuthoredOn = prescription.Date.Date,
+                        ResourceType = HiType.MedicationRequest,
+                        DosageInstruction = new DosageInstruction(1, prescription.Dosage),
+                        MedicationReference = new MedicationReference("Medication/" + medicationId, "Medications")
+                    }
+                };
+                var coding =
+                    new Coding(prescription.GenName, prescription.Medicine);
+                var medicationRepresentation = new MedicationRepresentation
+                {
+                    FullUrl = uuidMedication,
+                    Resource = new MedicationResource
+                    {
+                        Id = medicationId, ResourceType = HiType.Medication,
+                        Code = new Code(new List<Coding> {coding}, prescription.Medicine)
+                    }
+                };
+
+                list.Add(medicationRequestRepresentation);
+                list.Add(medicationRepresentation);
+            }
+
+            if (!list.Any()) return Option.None<MedicationResponse>();
+            var medicationResponse = new MedicationResponse
+            {
+                Entry = list,
+                Id = Uuid.Generate().Value.Split(":").Last(),
+                ResourceType = "Bundle",
+                Type = "collection"
+            };
+            return Option.Some(medicationResponse);
+        }
+
+        private static async Task<Option<ObservationResponse>> FetchClinicalNotes(DataRequest dataRequest,
+            List<ClinicalNote> clinicalNotes, string patientName)
+        {
+            LogDataRequest(dataRequest);
+            var uuid = Uuid.Generate().Value;
+            var id = uuid.Split(":").Last();
+            var observations = (from clinicalNote in clinicalNotes
+                where WithinRange(dataRequest.DateRange, clinicalNote.CreatedDate)
+                select new ObservationRepresentation
+                {
+                    FullUrl = uuid,
+                    Resource = new Resource(HiType.Observation, id, "final", new Code("Clinical notes"),
+                        new Subject(patientName), new List<Performer> {new Performer(clinicalNote.UserName)},
+                        clinicalNote.CreatedDate, clinicalNote.Note),
+                }).ToList();
+
+            if (!observations.Any()) return Option.None<ObservationResponse>();
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)